File: app/src/main/java/com/example/renqingyou/noandroidxfragment/TestFragment.java
package com.example.renqingyou.noandroidxfragment;

//import android.app.Fragment;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.AppCompatButton;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.sensorsdata.analytics.android.sdk.SensorsDataAPI;
import com.sensorsdata.analytics.android.sdk.SensorsDataFragmentTitle;
import com.sensorsdata.analytics.android.sdk.SensorsDataTrackFragmentAppViewScreen;

@SuppressWarnings("Convert2Lambda")
public class TestFragment extends Fragment {
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_test, container, false);
        AppCompatButton button = view.findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(view.getContext(),"我是fragment上的button",Toast.LENGTH_SHORT).show();
                SensorsDataAPI.sharedInstance().trackViewScreen(TestFragment.this);
            }
        });
        return view;
    }

    @Override
    public void onResume() {
        super.onResume();
        //SensorsDataAPI.sharedInstance().trackViewScreen(this);
    }

    @Override
    public void setUserVisibleHint(boolean isVisibleToUser) {
        super.setUserVisibleHint(isVisibleToUser);
    }

    @Override
    public void onHiddenChanged(boolean hidden) {
        super.onHiddenChanged(hidden);
    }

    @Override
    public boolean getUserVisibleHint() {
        return super.getUserVisibleHint();
    }
}

File: app/src/main/java/com/example/renqingyou/noandroidxfragment/TestFragment.kt
package com.example.renqingyou.noandroidxfragment

//import android.app.Fragment;
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.AppCompatButton
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast

import com.sensorsdata.analytics.android.sdk.SensorsDataAPI
import com.sensorsdata.analytics.android.sdk.SensorsDataTrackFragmentAppViewScreen


@SensorsDataTrackFragmentAppViewScreen
class TestFragment : Fragment() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_test, container, false)
        val button = view.findViewById<AppCompatButton>(R.id.button)
        button.setOnClickListener { view ->
            Toast.makeText(view.context, "我是fragment上的button", Toast.LENGTH_SHORT).show()
            //SensorsDataAPI.sharedInstance().trackViewScreen();
        }
        return view
    }

    override fun onResume() {
        super.onResume()
        //SensorsDataAPI.sharedInstance().trackViewScreen(this);
    }

    override fun setUserVisibleHint(isVisibleToUser: Boolean) {
        super.setUserVisibleHint(isVisibleToUser)
    }

    override fun onHiddenChanged(hidden: Boolean) {
        super.onHiddenChanged(hidden)
    }
}
